pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -135,11 +149,12 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 		}
 	}
 	var newBuild *buildapi.Build
-	if isBuild {
+	switch resource {
+	case "builds":
 		if newBuild, err = client.Builds(namespace).Clone(request); err != nil {
 			return err
 		}
-	} else {
+	case "buildconfigs":
 		if newBuild, err = client.BuildConfigs(namespace).Instantiate(request); err != nil {
 			return err
 		}
```

Comment:
We already have that covered in L126, but uf you really want to check it here too, I can add it.